types/redux-form/lib/FieldArray.d.ts

```diff
@@ -10,7 +10,7 @@ interface _BaseFieldArrayProps<P = {}, FieldValue = any> {
     rerenderOnEveryChange?: boolean | undefined;
 }
 
-export type BaseFieldArrayProps<P = {}, FieldValue = any> = (P | { props: P}) & _BaseFieldArrayProps<P, FieldValue>;
```

Comment:
I'm pretty unfamiliar with these types, but the current behavior definitely looks intentional.  In some of the tests under the one you added, you can see that two approaches are used - merging properties into the parent or adding them under a new `props` property.

If I'm reading 7c28be5245fa6959a8f587e1410ef12c261b02d8 correctly, it looks like the need to insert an ugly `<{}>` comes from a type inference failure and not from a problem with the types.

Personally, my inclination would be to leave things as they are, though I'm open to feedback, particularly from the owners of these types.

Having said that, if we go ahead with a fix like this, it seems like it should be `P & { props?: P }` to match (what I assume is a workaround in) `Field.d.ts`.

Reply:
After so long, I also can't tell what I was talking about in `Field.d.ts`. 😅 

As I said before, I think the current types are correct and this is a workaround for a TS issue (which is probably already known and, given that it's so old, very difficult to fix).  However, I realize that that doesn't help you and your change seems acceptable.